src/settings/OrderNumber.js

```diff
@@ -22,44 +14,61 @@ class OrderNumber extends Component {
 
   constructor(props) {
     super(props);
+
     this.configManager = props.stripes.connect(ConfigManager);
   }
 
   getInitialValues = (settings) => {
-    const value = settings.length && settings[0].value === 'true';
+    const value = settings.length === 0 ? '' : settings[0].value;
+    const defaultConfig = {
+      orderNumber: false,
+      selectedPrefixes: [],
+      prefixesList: [],
+      selectedSuffixes: [],
+      suffixesList: [],
+    };
+    let config;
+
+    try {
+      config = Object.assign({}, defaultConfig, JSON.parse(value));
+    } catch (e) {
+      config = defaultConfig;
+    }
+
+    return config;
+  }
+
+  beforeSave = (data) => {
+    const {
+      orderNumber,
+      selectedPrefixes,
+      prefixesList,
+      selectedSuffixes,
+      suffixesList,
+    } = data;
 
-    return { orderNumber: value };
+    return JSON.stringify({
+      orderNumber,
+      selectedPrefixes,
```

Comment:
I hope you're sending to back-end a plain lists of strings? like `['OO', 'OO2']`